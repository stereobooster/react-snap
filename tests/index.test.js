const nativeFs = require("fs");

const rewiremock = require("rewiremock").default;
rewiremock("mkdirp").with({
  sync: (...args) => console.log(args)
});

const DevNullStream = require("dev-null-stream");
const devNullStream = new DevNullStream();

const cwd = process.cwd();
const writeFileSyncMock = jest.fn();
const fs = {
  existsSync: nativeFs.existsSync,
  createReadStream: nativeFs.createReadStream,
  createWriteStream: (...args) => devNullStream,
  writeFileSync: (path, content) => {
    writeFileSyncMock(path.replace(cwd, ""), content);
  }
};

const { run } = require("./../index.js");

rewiremock.enable();

test("one page test", async () => {
  await run(
    {
      source: "tests/examples/one-page",
      puppeteerArgs: ['--no-sandbox', '--disable-setuid-sandbox']
    },
    {
      fs
    }
  );
  expect(writeFileSyncMock.mock.calls.length).toEqual(1);
  expect(writeFileSyncMock.mock.calls).toMatchSnapshot();
});
