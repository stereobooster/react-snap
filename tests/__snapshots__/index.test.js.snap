// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`one page proceeds index.html 1`] = `"<html lang=\\"en\\"><head><meta charset=\\"utf-8\\"></head><body><script>document.body.appendChild(document.createTextNode(\\"test\\"));</script>test</body></html>"`;
